apps/web/src/components/Composer/NewPublication.tsx

```diff
@@ -907,39 +1025,81 @@ const NewPublication: FC<NewPublicationProps> = ({ publication }) => {
         </Wrapper>
       ) : null}
       <div className="block items-center px-5 sm:flex">
-        <div className="flex items-center space-x-4">
-          <Attachment />
-          <Gif setGifAttachment={(gif: IGif) => setGifAttachment(gif)} />
-          {!publication?.isDataAvailability && (
-            <>
-              <CollectSettings />
-              <ReferenceSettings />
-              <AccessSettings />
-            </>
-          )}
-          <PollSettings />
-        </div>
-        <div className="ml-auto pt-2 sm:pt-0">
-          <Button
-            disabled={
-              isLoading ||
-              isUploading ||
-              isSubmitDisabledByPoll ||
-              videoThumbnail.uploading
-            }
-            icon={
-              isLoading ? (
-                <Spinner size="xs" />
-              ) : isComment ? (
-                <ChatAlt2Icon className="h-4 w-4" />
-              ) : (
-                <PencilAltIcon className="h-4 w-4" />
-              )
-            }
-            onClick={createPublication}
-          >
-            {isComment ? t`Comment` : t`Post`}
-          </Button>
+        {showNewPublicationModal &&
+        modalPublicationType === NewPublicationTypes.Spaces ? (
+          <SpaceSettings />
+        ) : (
+          <div className="flex items-center space-x-4">
+            <Attachment />
+            <Gif setGifAttachment={(gif: IGif) => setGifAttachment(gif)} />
+            {!publication?.isDataAvailability && (
+              <>
+                <CollectSettings />
+                <ReferenceSettings />
+                <AccessSettings />
+              </>
+            )}
+            <PollSettings />
+          </div>
+        )}
+        <div className="absolute bottom-2 right-2 inline-flex">
+          <div className="ml-auto pt-2 sm:pt-0">
+            <Button
+              disabled={
+                isLoading ||
+                isUploading ||
+                isSubmitDisabledByPoll ||
+                videoThumbnail.uploading
+              }
+              icon={getButtonIcon({
+                isLoading,
+                isComment,
+                showNewPublicationModal,
+                modalPublicationType
+              })}
+              onClick={createPublication}
+            >
+              {getButtonText({
+                isComment,
+                showNewPublicationModal,
+                modalPublicationType
+              })}
```

Comment:
I think if we move this function inside component we can reduce all those params? no?